src/test/java/com/endava/internship/mocking/service/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,155 @@
+package com.endava.internship.mocking.service;
+
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.repository.InMemPaymentRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class InMemPaymentRepositoryTest {
+
+    private InMemPaymentRepository paymentRepository;
+
+    private final static List<Payment> payments = Stream.of(
```

Comment:
don't forget to use capital leters to show that this is a constant, it easier in code to undersand that this variable should be the same permanently. Btw did you hear something about effectivly final in java?

Reply:
Good, fixed.  A variable is considered an effective final if it is not modified after initialization in the local block. This saves us from initializing final variables when we want to use them in nested classes, or lambdas.